backend/test_observer/controllers/test_results/test_results.py

```diff
@@ -0,0 +1,307 @@
+) -> TestResultSearchResponse:
+    """
+    Search test results across artefacts using flexible filters.
+
+    This endpoint uses a single optimized query with window functions to get both
+    the total count and paginated results in one database round trip.
+    """
+
+    # Build filters
+    filters = []
+
+    if family:
+        try:
+            family_enum = FamilyName(family.lower())
+            filters.append(Artefact.family == family_enum)
+        except ValueError:
+            filters.append(Artefact.family.is_(None))
+
+    if environment:
+        filters.append(Environment.name.ilike(f"%{environment}%"))
+
+    if test_case:
+        filters.append(TestCase.name.ilike(f"%{test_case}%"))
+
+    if template_id:
+        filters.append(TestCase.template_id == template_id)
+
+    if issues:
+        # Handle multiple issue IDs separated by commas
+        issue_list = [issue.strip() for issue in issues.split(",") if issue.strip()]
+        if issue_list:
+            try:
+                issue_ids = [int(issue_id) for issue_id in issue_list]
+                filters.append(
+                    TestResult.id.in_(
+                        select(IssueTestResultAttachment.test_result_id)
+                        .join(IssueTestResultAttachment.issue)
+                        .where(Issue.id.in_(issue_ids))
+                    )
+                )
+            except ValueError:
+                return TestResultSearchResponse(count=0, test_results=[])
+
+    if from_date:
+        filters.append(TestExecution.created_at >= from_date)
+
+    if until:
+        filters.append(TestExecution.created_at <= until)
+
+    # Build the main query with window function for total count in a single query
+    query = (
+        select(TestResult, func.count().over().label("total_count"))
+        .join(TestResult.test_execution)
+        .join(TestExecution.environment)
+        .join(TestExecution.artefact_build)
+        .join(ArtefactBuild.artefact)
+        .join(TestResult.test_case)
+        .options(
+            joinedload(TestResult.test_execution).joinedload(TestExecution.environment),
+            joinedload(TestResult.test_execution)
+            .joinedload(TestExecution.artefact_build)
+            .joinedload(ArtefactBuild.artefact),
+            joinedload(TestResult.test_case),
+        )
+        .order_by(desc(TestExecution.created_at), desc(TestResult.id))
+        .offset(offset)
+        .limit(limit)
+    )
+
+    # Apply all filters
+    if filters:
+        query = query.where(and_(*filters))
+
+    result = db.execute(query).all()
+
+    if result:
+        test_results = [row[0] for row in result]
+        total_count = result[0][1]
+    else:
+        test_results = []
+        total_count = 0
+
+    return TestResultSearchResponse(count=total_count, test_results=test_results)
+
+
+@router.get("/families", response_model=list[str])
+def get_families(db: Session = Depends(get_db)) -> list[str]:
```

Comment:
Not sure if we need this endpoint, this enum is sort of hardcoded on the frontend already